### PR TITLE
(feat) RDCC-2230 : Updating the required roles for FetchUsersById.

### PR DESCRIPTION
**Pull request by nitishsingla1234**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-2230

### Change description ###
(feat) RDCC-2230 : Updating the required roles for FetchUsersById.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by shashank-rastogi — DISMISSED** [review]

**Review by AdilOoze — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java [hunk 1]
```diff
@@ -256,7 +256,7 @@
             consumes = APPLICATION_JSON_VALUE,
             produces = APPLICATION_JSON_VALUE
     )
-    @Secured("cwd-admin")
+    @Secured({"cwd-system-user"})
```
Comment (SujeethKumarAri) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java:259]: The ticket says to add below role as well
`cwd-read-only`. Correct me if I am wrong

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -91,7 +91,7 @@ idam:
     microservice: rd_caseworker_ref_api
     url: ${S2S_URL:http://rpe-service-auth-provider-aat.service.core-compute-aat.internal}
   s2s-authorised:
-    services: ${CRD_S2S_AUTHORISED_SERVICES:rd_caseworker_ref_api,rd_professional_api,am_org_role_mapping_service}
+    services: ${CRD_S2S_AUTHORISED_SERVICES:rd_caseworker_ref_api,rd_professional_api,am_org_role_mapping_service,iac}
```
Comment (shreedhar-hmcts) [inline src/main/resources/application.yaml:94]: added iac in flux ?

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java [hunk 3]
```diff
@@ -256,7 +256,7 @@
...
     public ResponseEntity<Object> fetchCaseworkersById(@RequestBody UserRequest userRequest) {
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java:260]: required 2 roles ?
'cwd-read-only' role for the IAC service
'cwd-system-user' role for the ORM service (ORM service name am_org_role_mapping_service)

File: src/main/resources/application.yaml (hunk 2 above)
Comment (nitishsingla1234) [reply src/main/resources/application.yaml:94]: @shreedhar-hmcts I couldn't see flux for crd. Can you please confirm if it exists.
I had the similar doubt as well.

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java (hunk 3 above)
Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java:260]: For now, we need the system user role only.

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java (hunk 1 above)
Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/controllers/CaseWorkerRefController.java:259]: For now, we need the system user role only.

File: src/main/resources/application.yaml (hunk 2 above)
Comment (nitishsingla1234) [reply src/main/resources/application.yaml:94]: Added Flux config in PR https://github.com/hmcts/cnp-flux-config/pull/7790

Comment (shreedhar-hmcts) [reply src/main/resources/application.yaml]: Good

**Review by shreedhar-hmcts — APPROVED** [review]

**Review by shashank-rastogi — APPROVED** [review]

**Pull request merged**